server.go

```diff
@@ -33,55 +36,88 @@ func (s *Server) Listen() {
 	for {
 		conn, err := s.l.Accept()
 		if err != nil {
-			log.Fatal(err)
+			s.logf("golfish: failed to accept incomming connection: %v", err)
+			continue
 		}
 
-		go s.handleConn(conn)
+		go func() {
+			if err := s.handleConn(conn); err != nil {
+				s.logf("goldfish: unable to handle request from %v: %v", conn.RemoteAddr(), err)
+
+			}
+			if err := conn.Close(); err != nil {
+				s.logf("goldfish: failed to close connection with %v: %v", conn.RemoteAddr(), err)
+			}
+		}()
 	}
 }
 
-func (s *Server) handleConn(conn net.Conn) {
-	defer func() {
-		if err := conn.Close(); err != nil {
-			log.Printf("Failed to  close connection with client: %v", err)
-		}
-	}()
+func (s *Server) handleConn(conn io.ReadWriteCloser) error {
 	r := bufio.NewReader(conn)
+	for {
+		buf, err := s.readMessage(r)
+		if err != nil {
+			return fmt.Errorf("failed to read message from connection: %v", err)
```

Comment:
Error message is inconsistent compared to error messages on line 39, 45 en 49 (not starting with "goldfish").